tests/Unit/PointwiseFunctions/Hydro/EquationsOfState/Test_PiecewisePolytropicFluid.cpp

```diff
@@ -0,0 +1,491 @@
+  }
+}
+
+template <bool IsRelativistic>
+void check_edge_cases() {
+  // Two edge cases checked are:
+  // * The piecewise polytrope (with identical exponents) matches the single
+  // polytrope case
+  //
+  // * The rest mass density at the transition density matches using both the
+  //  high and low density constants/exponents
+
+  const double transition_density = 10.0;
+  const double polytropic_constant = 0.5;
+  const double polytropic_exponent = 1.5;
+
+  const Scalar<double> rest_mass_density_high{1.1 * transition_density};
+  const Scalar<double> rest_mass_density_transition{transition_density};
+  const Scalar<double> rest_mass_density_low{0.9 * transition_density};
+
+  const auto eos = EquationsOfState::PiecewisePolytropicFluid<IsRelativistic>{
+      transition_density, polytropic_constant, 0.9 * polytropic_exponent,
+      1.1 * polytropic_exponent};
+
+  // At the transition density, this EoS defaults to using the high density
+  // constants and exponents to calculate remaining thermodynamic quantities.
+  // Here check that eint matches at the transition density using low density
+  // constants and exponents as well.
+  const double transition_pressure =
+      polytropic_constant * pow(transition_density, 0.9 * polytropic_exponent);
+
+  const double transition_spec_eint_low_params =
+      transition_pressure /
+      ((0.9 * polytropic_exponent - 1.0) * transition_density);
+
+  CHECK(get(eos.specific_internal_energy_from_density(
+            rest_mass_density_transition)) ==
+        approx(transition_spec_eint_low_params));
+
+  // Compare single and piecewise polytrope with same parameters
+  const auto eos_piecewise_polytrope =
+      EquationsOfState::PiecewisePolytropicFluid<IsRelativistic>{
+          transition_density, polytropic_constant, polytropic_exponent,
+          polytropic_exponent};
+  const auto eos_single_polytrope =
+      EquationsOfState::PolytropicFluid<IsRelativistic>{polytropic_constant,
+                                                        polytropic_exponent};
+  // Compare single and identical piecewise polytrope
+  CHECK(get(eos_piecewise_polytrope.specific_internal_energy_from_density(
+            rest_mass_density_low)) ==
+        get(eos_single_polytrope.specific_internal_energy_from_density(
+            rest_mass_density_low)));
+  CHECK(get(eos_piecewise_polytrope.specific_internal_energy_from_density(
+            rest_mass_density_high)) ==
+        get(eos_single_polytrope.specific_internal_energy_from_density(
+            rest_mass_density_high)));
+  CHECK(get(eos_piecewise_polytrope.chi_from_density(rest_mass_density_low)) ==
+        get(eos_single_polytrope.chi_from_density(rest_mass_density_low)));
+  CHECK(get(eos_piecewise_polytrope.chi_from_density(rest_mass_density_high)) ==
+        get(eos_single_polytrope.chi_from_density(rest_mass_density_high)));
+}
+
+void check_dominant_energy_condition_at_bound() {
+  const auto seed = std::random_device{}();
```

Comment:
You can just not specify the `seed`. We only have that functionality so that if the code fails in CI we can play back a specific seed to reproduce the failure :)